packages/frontend/src/components/accounts/recovery_setup/new_account/SetupPassphraseNewAccount.js

```diff
@@ -0,0 +1,76 @@
+import { KeyPair } from 'near-api-js';
+import { generateSeedPhrase } from 'near-seed-phrase';
+import React, { useEffect, useState } from 'react';
+
+import { Mixpanel } from '../../../../mixpanel/index';
+import ConfirmPassphrase from './ConfirmPassphrase';
+import SavePassphrase from './SavePassphrase';
+
+export default ({ handleConfirmPassphrase }) => {
+    const [passPhrase, setPassPhrase] = useState('');
+    const [recoveryKeyPair, setRecoveryKeyPair] = useState();
+    const [implicitAccountId, setImplicitAccountId] = useState('');
+
+    const [confirmPassphrase, setConfirmPassphrase] = useState(false);
+    const [wordId, setWordId] = useState(null);
+    const [userInputValue, setUserInputValue] = useState('');
+    const [userInputValueWrongWord, setUserInputValueWrongWord] = useState(false);
+
+    const generateAndSetPhrase = () => {
+        const { seedPhrase, secretKey } = generateSeedPhrase();
+        const recoveryKeyPair = KeyPair.fromString(secretKey);
+        const implicitAccountId = Buffer.from(recoveryKeyPair.publicKey.data).toString('hex');
+
+        setPassPhrase(seedPhrase);
+        setRecoveryKeyPair(recoveryKeyPair);
+        setImplicitAccountId(implicitAccountId);
+        setWordId(Math.floor(Math.random() * 12));
+    };
+
+    useEffect(() => {
+        generateAndSetPhrase();
+    }, []);
+
+    if (confirmPassphrase) {
+        return (
+            <ConfirmPassphrase
+                wordId={wordId}
+                userInputValue={userInputValue}
+                userInputValueWrongWord={userInputValueWrongWord}
+                handleChangeWord={(userInputValue) => {
```

Comment:
<3 just a note that I really like the choice of in-lining most event handlers here rather than making them functions in the closure scope :). It made it very apparent that `generateAndSetPhrase` was a special case that is re-used